dev/com.ibm.ws.transport.http/src/com/ibm/wsspi/genericbnf/HeaderKeys.java

```diff
@@ -120,17 +122,35 @@ protected void setUndefined(boolean flag) {
 
     /**
      * Get the enumerated instance that matches this input ordinal.
-     * 
+     *
      * @param i
      * @return Object (HeaderKeys)
      * @throws IndexOutOfBoundsException
-     *             if this ordinal is not valid
+     *                                       if this ordinal is not valid
      */
     public abstract Object getEnumByOrdinal(int i);
 
+    /**
+     * Set keysUsed true - means the HeaderKey has been used in a request or response
+     *
+     * @param ordinal
+     */
+    public void setKeyUsed() {
+        keyUsed = true;
```

Comment:
For this to work you would need a guarantee that everyone is using the same instance of the HeaderKeys object or you would need to store this in a set instead of as an instance variable.  You could guarantee that everyone is using the same instance by having a Map<HeaderKeys, HeaderKeys> to relatively intern the HeaderKeys objects.   The other map option I was saying is having a Set is to have a concurrent Set that you insert the HeaderKeys.getName() into the set.  The main concern of course is memory leak.